data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -581,8 +581,14 @@ fi
 # Check if the API and API_INT Server URLs can be reached.
 echo "Check if API and API-Int URLs are reachable during bootstrap"
 
-check_url "API_URL" "${API_SERVER_URL}"
-check_url "API_INT_URL" "${API_INT_SERVER_URL}"
+if check_url "API_URL" "${API_SERVER_URL}"; then
+  echo "API_URL ${SERVER_URL} is not ready"
+  exit 1
```

Comment:
I'm not sure we should `exit` here, as I don't think we want the check to fail the script, just to warn users about this potential issue.

Reply:
Don't we want to gate bootstrap until the API servers are available on the master nodes?

Reply:
Oh, perhaps. These checks were initially added by the installer team to help users debug failures if, for some reason, the private dns for the cluster failed to set up properly. For that case, we did not want this debugging tool to affect the installation process at all, just collect data for the user. 

Gating cluster bootstrapping sounds reasonable to me, and I'm fine with using this function too, but still catching up on understanding these particular details.